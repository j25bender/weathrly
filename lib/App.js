import React from 'react';
import CurrentWeather from '../lib/CurrentWeather.js';
import SevenHourForecast from '../lib/SevenHourForecast.js';
import Welcome from '../lib/Welcome.js';
import key from '../lib/weatherUndergroundAPIKey.js';
import cleanData from '../lib/cleanData.js';


 export default class App extends React.Component {
    constructor() {
        super();

    this.state = {
        CurrentWeather: {},
        SevenHourForecast: [],
        TenDayForecast: []
    }
    this.fetchData = this.fetchData.bind(this);    
}

    render() {
        return (
            <div>
                <Welcome searchValue={this.fetchData}/>
                <CurrentWeather forecast={this.state.CurrentWeather}/>
                <SevenHourForecast forecast={this.state.SevenHourForecast}/>
                {/* <TenDayForecast forecast={this.state.TenDayForecast}/> */}
            </div>
        )
    }

    testInput(value) {
        if (isNaN(parseInt(value))) {
            value = value.split(", ");
            return `${value[1]}/${value[0]}`;
        } else {
            return value;
        }
    }

    componentDidMount() {
        let location = localStorage.getItem("location");
        console.log(location)
    
        if (location !== null) {
            this.fetchData(location);
        }
    }

    fetchData(inputValue) {
        this.testInput(inputValue);
        fetch(`http://api.wunderground.com/api/${key}/conditions/forecast10day/hourly/q/${inputValue}.json`)
        .then( response => response.json() )
        .then( json => {
            let filteredData = cleanData(json)
            this.setState({
                CurrentWeather: filteredData.CurrentObject,
                SevenHourForecast: filteredData.sevenHourForecastArray,
                TenDayForecast: filteredData.tenDayArray
            })
        })
    }
}