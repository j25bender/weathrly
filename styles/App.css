body {
    background-image: radial-gradient(
      farthest-corner at 2rem 2rem,
      #42aaf4 0%,
      #0a2963 100%
    );
    box-sizing: border-box;
    font-family: "Open Sans", sans-serif;
    margin: auto;
    padding: 2rem;
    text-align: center;
}