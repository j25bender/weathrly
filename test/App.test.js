import React from 'react';
import { shallow, mount } from 'enzyme';
import App from '../lib/App';
import cleanData from '../lib/cleanData';
import mockData from '../lib/mockData';
import 'jest-localstorage-mock';

describe('App Tests', () => {
  let filteredData;
  let wrapper;
  
  beforeEach(() => {
      filteredData = cleanData(mockData);
      wrapper = mount(<App />);
  });
  
  it('App should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('Should be a function', () => {
    expect(typeof wrapper.state).toEqual('function');
  });

  it('App should mount <Welcome />', () => {   
     expect(wrapper.find('h1').text()).toEqual('Welcome to Weathrly!');
     expect(App).toMatchSnapshot();
  });

  it('Should render Welcome component when it does not have any data in localStorage.', () => {
    global.localStorage.abc = null;
    const wrapper = mount(<App />);
    expect(wrapper.find('CurrentWeather').length).toEqual(0);
    expect(wrapper.find('Welcome').length).toEqual(1);
  });

  it('Should render the weather page when a city is specified', () => {  
    expect(wrapper.state()).toEqual({ CurrentWeather: {},                                                                                        SevenHourForecast: [],                                                                                     TenDayForecast: []
                                    })

    expect(wrapper.state().CurrentWeather.low).toBeUndefined();
    expect(wrapper.state().SevenHourForecast.length).toEqual(0);
    expect(wrapper.state().TenDayForecast.length).toEqual(0)
                                    
    wrapper.setState({CurrentWeather: filteredData.CurrentObject,
                      SevenHourForecast: filteredData.sevenHourForecastArray,
                      TenDayForecast: filteredData.tenDayArray
                     })
                        
    expect(wrapper.state().CurrentWeather.low).toEqual('32');
    expect(wrapper.state().SevenHourForecast.length).toEqual(7);
    expect(wrapper.state().TenDayForecast.length).toEqual(10);  

    expect(App).toMatchSnapshot();    
  });
});
