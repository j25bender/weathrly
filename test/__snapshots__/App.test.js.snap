// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Tests App should mount <Search /> 1`] = `[Function]`;

exports[`App Tests Should render the weather page when a city is specified 1`] = `[Function]`;
